### PR TITLE
Add: (partially) Connecting Payment Form

### PR DESCRIPTION
**Pull request by RobinCsl**
Displaying Card image conditionally on card number

@tbergq I believe I should add tests too, so I'll tag it WIP

### CODE REVIEW

File: app/MMB/src/scenes/tripServices/insurance/paymentScene/CardImage.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+type Props = {|
+  +cardType: CardType,
+|};
+
+const CardImage = ({ cardType }: Props) => {
+  let source;
+  switch (cardType) {
+    case 'MASTERCARD': {
+      source = require('./images/MasterCardLogo.imageset/mastercard.png');
+      break;
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/insurance/paymentScene/CardImage.js]: Does this really work?

https://github.com/facebook/react-native/issues/6391#issuecomment-194568185

File: app/MMB/src/scenes/tripServices/insurance/paymentScene/InsurancePaymentContext.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+});
+
+type Props = {|
+  +children: React.Node,
+|};
+
+type State = {|
+  cardNumber: string,
+  cardType: string,
+  expiryDate: { month: string, year: string },
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/insurance/paymentScene/InsurancePaymentContext.js]: should also be strict type

File: app/MMB/src/scenes/tripServices/insurance/paymentScene/InsurancePaymentContext.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  securityCode: string,
+  cardHoldersName: string,
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/insurance/paymentScene/InsurancePaymentContext.js]: `cardholderName` maybe?

File: app/MMB/src/scenes/tripServices/insurance/paymentScene/InsurancePaymentContext.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  | 'MASTERCARD'
+  | 'VISA'
+  | 'MAESTRO'
+  | 'AMERICAN_EXPRESS'
+  | 'DISCOVER'
+  | 'DINERS_CLUB'
+  | 'MIR'
+  | 'UNKNOWN';
+
+function detectCardType(cardNumber: string, strict: boolean = false): CardType {
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/insurance/paymentScene/InsurancePaymentContext.js]: could you add some test for this?

File: app/MMB/src/scenes/tripServices/insurance/paymentScene/CardImage.js (hunk 1 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/insurance/paymentScene/CardImage.js]: Well, the image was changing when I tried different card numbers

**Comment by RobinCsl** [comment]
I added some tests to verify that it detects the right card. 
The current implementation is just for displaying the right image; it does not check if the card is valid.

**Pull request merged**